lib/Doctrine/DBAL/Cache/ResultCacheStatement.php

```diff
@@ -59,11 +61,12 @@ class ResultCacheStatement implements IteratorAggregate, ResultStatement
     private $defaultFetchMode = FetchMode::MIXED;
 
     /**
-     * @param string $cacheKey
-     * @param string $realKey
-     * @param int    $lifetime
+     * @param ResultStatement|PDOStatement $stmt
```

Comment:
I wanted to add an `assert` statement to the constructor to check for this, but apparently PHPStan thinks it's redundant. The code in question would be

```php
assert($stmt instanceof ResultStatement || $stmt instanceof PDOStatement);
```

Do we want to add it and silence the error or leave it as is?

Reply:
This should be removed and the type hint added back, isn't it?

Reply:
I agree here, it looks like we're missing an assert on the caller and not here.